shell/platform/darwin/macos/framework/Source/FlutterCompositor.mm

```diff
@@ -3,22 +3,30 @@
 // found in the LICENSE file.
 
 #import "flutter/shell/platform/darwin/macos/framework/Source/FlutterCompositor.h"
-#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterMutatorView.h"
 
 #include "flutter/fml/logging.h"
 
 namespace flutter {
 
 namespace {
-std::vector<PlatformViewLayerWithIndex> CopyPlatformViewLayers(const FlutterLayer** layers,
-                                                               size_t layer_count) {
-  std::vector<PlatformViewLayerWithIndex> platform_views;
+std::vector<LayerVariant> CopyLayers(const FlutterLayer** layers, size_t layer_count) {
+  std::vector<LayerVariant> layers_copy;
   for (size_t i = 0; i < layer_count; i++) {
-    if (layers[i]->type == kFlutterLayerContentTypePlatformView) {
-      platform_views.push_back(std::make_pair(PlatformViewLayer(layers[i]), i));
+    const auto& layer = layers[i];
+    if (layer->type == kFlutterLayerContentTypePlatformView) {
+      layers_copy.push_back(PlatformViewLayer(layer));
+    } else if (layer->type == kFlutterLayerContentTypeBackingStore) {
+      std::vector<FlutterRect> rects;
+      auto present_info = layer->backing_store_present_info;
+      if (present_info) {
+        for (size_t j = 0; j < present_info->paint_region->rects_count; j++) {
+          rects.push_back(present_info->paint_region->rects[j]);
+        }
```

Comment:
Consider pre-reserving size of rects, and using std::copy which might have better performance:
```cc
rects.reserve(present_info->paint_region->rects_count);
std::copy(present_info->paint_region->rects,
          present_info->paint_region->rects + present_info->paint_region->rects_count,
          back_inserter(rects));
```